src/section.rs

```diff
@@ -105,47 +128,49 @@ impl Section {
             }
         }
 
-        if self.prefix == Prefix::EMPTY {
-            // If we are the root section (our prefix is empty), bump everyone's
-            // (except the new node) age by one.
-            for node in self.nodes.values_mut() {
-                node.increment_age()
-            }
+        // During startup, nodes joining as adult (age of 5), and no relocation.
+        let startup = self.prefix == Prefix::EMPTY;
+
+        let new_node = if startup {
+            Node::new(node.name(), params.adult_age)
         } else if node.is_infant(params) &&
                    node::count_infants(params, self.nodes.values()) >=
                        params.max_infants_per_section
         {
             return self.reject_node(node);
-        }
+        } else {
+            node
+        };
 
-        let name = node.name();
-        let is_adult = node.is_adult(params);
+        let age = new_node.age();
+        let name = new_node.name();
+        let is_adult = new_node.is_adult(params);
 
-        self.add_node(node);
-        self.update_elders(params);
+        self.add_node(new_node);
+        // A relocated adult shall only trigger relocate once.
+        // It's promotion shall not trigger relocation.
+        let _ = self.update_elders(params, false);
 
         let responses = self.try_split(params);
         if !responses.is_empty() {
             responses
-        } else if is_adult {
-            self.try_relocate(params, Some(name))
+        } else if is_adult && !startup {
+            self.try_relocate(params, Block::new(Event::Live, name, age))
         } else {
             Vec::new()
         }
     }
 
     fn handle_dead(&mut self, params: &Params, name: Name) -> Vec<Response> {
         if let Some(node) = self.drop_node(name) {
-            self.update_elders(params);
-
-            let responses = self.try_merge(params);
-            if !responses.is_empty() {
-                responses
-            } else if node.is_adult(params) {
-                self.try_relocate(params, None)
-            } else {
-                Vec::new()
+            let mut responses = self.update_elders(params, true);
+            responses.extend(self.try_merge(params));
+            if node.is_adult(params) {
+                if let Some(last_live) = self.chain.last_live() {
+                    responses.extend(self.try_relocate(params, last_live));
```

Comment:
If merge has been triggered, should we still trigger relocation?

Reply:
if we are the section triggering merge, in `try_relocate`, the number of nodes will be checked as well, and there will be no relocation happen.
if we are the section being told to merge, our nodes is already empty, so relocation won't happen either.

so, no need to change code?

Reply:
:+1: